core/job_exec.go

```diff
@@ -173,8 +173,8 @@ func BatchDelete(ctx context.Context, job *Job) *JobResponse {
 			// src is a glob
 			src := sources[0]
 			for obj := range client.List(ctx, src, true, storage.ListAllItems) {
-				if obj.Err != nil {
-					// TODO(ig): add proper logging
+				if err := obj.Err; err != nil {
+					log.Logger.Error("list: %v", err)
```

Comment:
Should we check context cancellation here? I am not sure if we should log them.

Reply:
Or, we can easily handle this on `storage.S3`  and not send cancellation errors to channel.

Reply:
since `storage.List` is async, we piggyback the errors onto the Objects transferred with the channel as you know. Currently if a user gets a 403 (no access to the bucket), we ignore it.

Reply:
Sorry, I could not explain it well. Let me clarify deeply:

Currently, each operation takes the context as a first parameter and we are bypassing each error to object channel on async List operations without checking its type ([ref](https://github.com/peak/s5cmd/blob/master/storage/s3.go#L209)). Since we are going to log them, I think we should ignore RequestCancelled errors. It can be easily done by using `IsCancelationError()` method on this [line] (https://github.com/peak/s5cmd/blob/master/storage/s3.go#L209) and not sending these errors to channel.

Following is the example usage:
```
^C2020/03/03 23:18:33 # Got signal, cleaning up...
-ERR list: RequestCanceled: request context canceled caused by: context canceled
```

Reply:
Hmm, right. I think all errors should propagate to the one who received it. Caller decides what to do with it. If the caller decides to log the error, we could suppress cancelation errors in the logger level. That'd require changing the logger.Error signature to receive `error` instead of `format, args...`.

Reply:
That is also another option and it makes sense for me to handle these errors on the logger level. We already need to break API for supporting structured logs (JSON etc.) so we can do it now :)

Reply:
Just sent a commit that wouldn't change the signature but i'm not 100% sure about it. Also we should change logger method names to take `f` suffix, as in `formatted`. logger.Error sounds like it only receives a string.

Reply:
I am also not sure. How about:
```
       if err != nil {
                  if !isCancellationError(err) {		
                          log.Logger.Error("storage: %v", err)
                  }
		return
	}
```

Reply:
Is this control on storage layer? It'd be better if we don't log here. If it's on the caller site, we need to add this check to multiple functions, right?

Reply:
It is on caller site. There will be duplicate code but we don't need to construct error on log package. Just a suggestion, we can keep current structure.

Reply:
Done.